### PR TITLE
Referral emails

### PR DESCRIPTION
**Pull request by camhashemi**
@ryanpbrewster @leogrim

### CODE REVIEW

File: kifi-backend/modules/shoebox/app/com/keepit/model/OrganizationMembershipRepo.scala [hunk 1]
```diff
@@ -199,8 +200,8 @@ class OrganizationMembershipRepoImpl @Inject() (
 
   def getByRole(orgId: Id[Organization], role: OrganizationRole, excludeState: Option[State[OrganizationMembership]] = Some(OrganizationMembershipStates.INACTIVE))(implicit session: RSession): Seq[Id[User]] = {
     excludeState match {
-      case Some(exclude) => (for { row <- rows if row.organizationId === orgId && row.state =!= exclude && row.role == role } yield row.userId).list
-      case None => (for { row <- rows if row.organizationId === orgId && row.role == role } yield row.userId).list
+      case Some(exclude) => (for { row <- rows if row.organizationId === orgId && row.state =!= exclude && row.role === role } yield row.userId).list
+      case None => (for { row <- rows if row.organizationId === orgId && row.role === role } yield row.userId).list
```
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/model/OrganizationMembershipRepo.scala:204]: :grin: nice catch, I wonder how long that query has been wrong.

File: kifi-backend/modules/shoebox/app/com/keepit/payments/CreditRewardCommander.scala [hunk 2]
```diff
@@ -72,7 +76,40 @@ class CreditRewardCommanderImpl @Inject() (
...
+  }
+
+  private def sendReferralCodeAppliedEmail(referrerOrgId: Id[Organization], referredOrgId: Id[Organization])(implicit session: RWSession): ElectronicMail = {
+    val referredOrg = orgRepo.get(referredOrgId)
+    val referrerOrg = orgRepo.get(referrerOrgId)
+    val adminEmails = orgMembershipRepo.getByRole(referrerOrgId, OrganizationRole.ADMIN).flatMap(adminId => Try(emailAddressRepo.getByUser(adminId)).toOption)
+    val subject = s"Your team's referral code was used by ${referredOrg.name} on Kifi"
+    val htmlBody =
+      s"""
+         |Your <a href="http://www.kifi.com/${referrerOrg.handle.value}">${referrerOrg.name}</a> team's referral code was used by <a href="http://www.kifi.com/${referredOrg.handle.value}">${referredOrg.name}</a>. If they upgrade to a standard plan on Kifi,
```
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/payments/CreditRewardCommander.scala]: `https`?

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/UserCommander.scala [hunk 3]
```diff
@@ -450,6 +450,13 @@ class UserCommanderImpl @Inject() (
 
                   Get started by visiting the page to <a href="http://www.kifi.com/teams/new">create a team</a>.
               """,
+            textBody = Some(
+              s"""
+                  Per your request, you can now create a team on Kifi from
```
Comment (andrewconner) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/UserCommander.scala]: Does this work? I thought margin needed the `|`'s

File: kifi-backend/modules/shoebox/app/com/keepit/payments/CreditRewardCommander.scala [hunk 4]
```diff
@@ -148,9 +188,42 @@ class CreditRewardCommanderImpl @Inject() (
     val evolvedReward = Reward(RewardKind.OrganizationReferral)(RewardKind.OrganizationReferral.Upgraded)(orgId)
     val crs = creditRewardRepo.getByReward(currentReward)
     assert(crs.size <= 1, s"Somehow there are multiple referral rewards for $orgId! $crs")
-    crs.map { crToEvolve =>
+    val rewards = crs.map { crToEvolve =>
       finalizeCreditReward(crToEvolve.withReward(evolvedReward), None)
     }
+    crs.headOption.foreach { crToEvolve =>
+      val referrerOrgId = accountRepo.get(crToEvolve.accountId).orgId
+      if (orgExpRepo.hasExperiment(orgId, OrganizationExperimentType.FAKE) && orgExpRepo.hasExperiment(referrerOrgId, OrganizationExperimentType.FAKE)) {
```
Comment (camhashemi) [inline kifi-backend/modules/shoebox/app/com/keepit/payments/CreditRewardCommander.scala:196]: added this guard until after QA in prod

File: kifi-backend/modules/shoebox/app/com/keepit/payments/CreditRewardCommander.scala [hunk 5]
```diff
@@ -148,9 +188,42 @@ class CreditRewardCommanderImpl @Inject() (
...
+    val subject = s"You earned a $$100 credit for ${referrerOrg.name} on Kifi"
+    val htmlBody =
+      s"""
+         |Your team, <a href="https://www.kifi.com/${referrerOrg.handle.value}">${referrerOrg.name}</a>, earned a $$100 credit from ${referredOrg.name}.
+         |We've added it to your <a href="https://www.kifi.com/${referrerOrg.handle.value}/settings/plan">team balance</a>. Thank you so much for spreading the word
+         |about Kifi with great teams like ${referredOrg.name}!
+       """.stripMargin
+    val textBody =
+      s"""
+         |Your team, ${referrerOrg.name}, earned a $$100 credit from ${referredOrg.name}. We've added it to your team balance.
```
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/payments/CreditRewardCommander.scala:216]: Can we change `$$100` to `orgReferrerCredit.toDollarString`?

File: kifi-backend/modules/shoebox/test/com/keepit/model/OrganizationMembershipRepoTest.scala [hunk 6]
```diff
@@ -106,5 +106,23 @@ class OrganizationMembershipRepoTest extends Specification with ShoeboxTestInjec
         sortedMembers === allMembers.sorted
       }
     }
+
+    "get members by role" in {
+      withDb() { implicit injector =>
+        val (org, owner, adminsExpected, membersExpected) = db.readWrite { implicit session =>
+          val owner = user().withName("Zyxwv", "Utsr").saved
+          val admins = Seq(user().withName("Aaron", "Aaronson"), user().withName("Barry", "Barnes")).map(_.saved)
+          val members = Seq(user().withName("Carl", "Carson"), user().withName("Carl", "Junior")).map(_.saved)
```
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/test/com/keepit/model/OrganizationMembershipRepoTest.scala:115]: Are these names necessary? If not, I think `UserFactory.users(3).saved` is easier to understand.

File: kifi-backend/modules/shoebox/app/com/keepit/payments/CreditRewardCommander.scala (hunk 5 above)
Comment (camhashemi) [inline kifi-backend/modules/shoebox/app/com/keepit/payments/CreditRewardCommander.scala:216]: :+1: good call

File: kifi-backend/modules/shoebox/test/com/keepit/model/OrganizationMembershipRepoTest.scala (hunk 6 above)
Comment (camhashemi) [inline kifi-backend/modules/shoebox/test/com/keepit/model/OrganizationMembershipRepoTest.scala:115]: not necessary, thanks.

**Pull request merged**